Scripts/RemediationScripts/Remediate-EnableSSLDBForMySQLFlexibleServer.ps1

```diff
@@ -0,0 +1,817 @@
+
+    $totalDBForMySQLFS = $(($validDBForMySQLFSDetails | Measure-Object).Count)
+
+    if ($totalDBForMySQLFS -eq 0) {
+        Write-Host "No Azure Database for MySQL flexible server(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$(($validDBForMySQLFSDetails|Measure-Object).Count)] Azure Database for MySQL flexible server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    $colsProperty = @{Expression = { $_.ResourceName }; Label = "ResourceName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceGroupName }; Label = "ResourceGroupName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.RequireSecureTransport }; Label = "RequireSecureTransport"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceId }; Label = "ResourceId"; Width = 100; Alignment = "left" }
+        
+    $validDBForMySQLFSDetails | Format-Table -Property $colsProperty -Wrap
+    
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetSSLForDBForMySQLFlexibleServer"
+
+    if (-not (Test-Path -Path $backupFolderPath)) {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }  
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "[Step 3 of 3] Set SSL to previous value on all Azure Database for MySQL flexible server(s) in the Subscription..."
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ( -not $Force) {        
+        Write-Host "Do you want to disable SSL for all Azure Database for MySQL flexible server(s) mentioned in the file?"  -ForegroundColor $([Constants]::MessageType.Warning)
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if ($userInput -ne "Y") {
+            Write-Host "SSL server parameter (require_secure_transport) will not be rolled back on Azure Database for MySQL flexible server(s) mentioned in the file. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host $([Constants]::DoubleDashLine)
+            return
+        }
+        Write-Host "Rolling back SSL server parameter (require_secure_transport) on Azure Database for MySQL flexible server(s) mentioned in the file..." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else {
+        Write-Host "'Force' flag is provided. Rolling back SSL server parameter (require_secure_transport) on Azure Database for MySQL flexible server(s) in the Subscription without any further prompts..." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
single dash line

Reply:
I have used single dash line for all the scenario below  at line 765